content/02.body.md

```diff
@@ -0,0 +1,282 @@
+This motivates the need for unsupervised methods that can prioritize potential relationships.
+Ideally, such methods should be able to generalize between extreme differences in degree distribution.
+
+### Feature-degree correlation
+
+Network analysis methods are varied and can examine networks at many scales, including node, node-pair, community, and whole-network.
+In this investigation, we restrict our attention to node-pair features used for purposes of link prediction, though the principles are also applicable to other types of network analyses.
+
+Bias in a network's degree distribution can be toward either higher or lower degree (Figure {@fig:node-degree-bias}).
+To make unbiased predictions, therefore, features used for prediction should not be strongly influenced by degree, either positively or negatively.
+In practice, though, many features commonly used for link prediction are correlated with degree (Figure {@fig:features-degree}).
+
+![Common link-prediction features are correlated with node degree. Shown are five features (Supplemental table 2) computed on literature-derived protein-protein interaction network [@doi:10.1093/nar/gky1131] and the geometric mean of the node pair's source and target degrees. All five features show a positive relationship with degree, though the magnitude of this correlation is highly variable. For example, preferential attachment and random walk with restart (RWR) appear to have stronger relationships with degree than the Jaccard index.](https://github.com/greenelab/xswap-analysis/raw/bd40eb244ffe37454dc08099f5742ef3e962c70d/img/2_link_prediction_feature_degree_correlation/without_prior.png){#fig:features-degree width="100%"}
+
+Some link prediction features are functions of basic node pair properties like degree.
+These features (Supplemental table 2) can allow the effect of node degree to be predicted or derived directly.
+For example, the resource allocation index is the sum of inverse degree of common neighbors between source and target nodes (in the symmetric case), while preferential attachment is the product of source and target degree [@doi:10.1140/epjb/e2009-00335-8; @doi:10.1145/1065385.1065415].
+Many features and link-prediction methods, however, cannot be interrogated to determine, _a priori_, the effect that node degree will have.
+Therefore, it is important to have a more general method for comparing the relative effects of node degree on features and other link-prediction methods.
+
+
+## Methods
+
+### Network permutation
+
+<!-- Need a discussion about what counterfactual would look like.
+Wouldn't expect zero features in "null condition".
+Should explain what the "null condition" should be, logically, and why
+permutations capture that.
+
+Also can't compare raw features between entities-->
+
+The raw values of features computed for a given network entity depend both on relevant and non-relevant information, including extraneous and biased structural properties of the network.
```

Comment:
What is the extraneous information that is not degree biased? Trying to remember from above